### PR TITLE
DTSCCI-145 Claim documents duplicating on claim documents tab

### PR DESCRIPTION
**Pull request by tiagreaney**
### JIRA link (if applicable) ###
(https://tools.hmcts.net/jira/browse/DTSCCI-145)

### Change description ###
Claim documents duplicating on claim documents tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceCallbackHandler.java [hunk 1]
```diff
@@ -393,9 +394,10 @@ private void assembleResponseDocuments(CaseData caseData, CaseData.CaseDataBuild
             List<Element<CaseDocument>> copy = assignCategoryId.copyCaseDocumentListWithCategoryId(
                     claimantUploads, DocCategory.DQ_APP1.getValue());
             if (Objects.nonNull(copy)) {
-                claimantUploads.addAll(copy);
+                dupeClaimantDefendantResponseDocs.addAll(copy);
             }
             updatedCaseData.claimantResponseDocuments(claimantUploads);
+            updatedCaseData.dupeClaimantDefendantResponseDocs(copy);
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceCallbackHandler.java:400]: I'd avoid using abbreviations

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 2]
```diff
@@ -267,6 +267,12 @@ public class CaseData extends CaseDataParent implements MappableObject {
     @Builder.Default
     private final List<Element<CaseDocument>> claimantResponseDocuments = new ArrayList<>();
 
+    @Builder.Default
+    private final List<Element<CaseDocument>> duplicateSystemGeneratedCaseDocs = new ArrayList<>();
+
+    @Builder.Default
+    private final List<Element<CaseDocument>> dupeClaimantDefendantResponseDocs = new ArrayList<>();
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:274]: same here, I'd avoid the "dupe" abbreviation

**Review by gmikerlemos — APPROVED** [review]
Just a few recommendations but, overall good to go

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/docmosis/GenerateResponseSealedSpec.java [hunk 3]
```diff
@@ -90,20 +90,19 @@ private CallbackResponse prepareSealedForm(CallbackParams callbackParams) {
                 caseData
             );
             assignCategoryId.assignCategoryIdToCaseDocument(stitchedDocument, DocCategory.DEF1_DEFENSE_DQ.getValue());
-            CaseDocument stitchedDocumentCopy = assignCategoryId.copyCaseDocumentWithCategoryId(
-                    stitchedDocument, DocCategory.DQ_DEF1.getValue());
+            CaseDocument stitchedDocumentCopy = assignCategoryId.copyCaseDocumentWithCategoryId(stitchedDocument, DocCategory.DQ_DEF1.getValue());
             if (nonNull(caseData.getRespondent2DocumentGeneration()) && caseData.getRespondent2DocumentGeneration().equals("userRespondent2")) {
```
Comment (rishikrsharma) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/docmosis/GenerateResponseSealedSpec.java:94]: ("userRespondent2").equals(caseData.getRespondent2DocumentGeneration())
replace this 2 check in 1 = equals check null then return false

**Review by rishikrsharma — DISMISSED** [review]

**Review by rishikrsharma — APPROVED** [review]

**Review by gmikerlemos — APPROVED** [review]

**Review by rishikrsharma — APPROVED** [review]

**Review by gmikerlemos — APPROVED** [review]

**Pull request merged**